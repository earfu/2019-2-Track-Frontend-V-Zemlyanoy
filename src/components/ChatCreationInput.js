import React from 'react';
import PropTypes from 'prop-types';
import chatDefaults from '../chatDefaults';

function ChatCreationInput({ onChange, onKeyPress, onSubmit, value }) {
	return (
		<input
			className="chat-name-input"
			type="text"
			value={value}
			onChange={onChange}
			placeholder={chatDefaults.chatName}
			onSubmit={onSubmit}
			onKeyPress={onKeyPress}
		/>
	);
}

ChatCreationInput.propTypes = {
	onChange: PropTypes.func.isRequired,
	onKeyPress: PropTypes.func.isRequired,
	onSubmit: PropTypes.func.isRequired,
	value: PropTypes.string.isRequired,
};

export default ChatCreationInput;
