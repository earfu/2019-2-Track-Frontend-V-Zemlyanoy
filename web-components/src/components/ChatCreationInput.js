const template = document.createElement('template');
template.innerHTML = `
    <style>
        input {
            border: 0;
            outline: none;
            width: calc(100% - 2px);
            background-color: #ffffcc;
        }

        :host {
            display: inherit;
            height: 100%;
            border: 1px solid rgba(25, 25, 25, 0.32);
            margin-bottom: 1px;
        }
    </style>
    <input type="text">
`;

export default class ChatCreationInput extends HTMLElement {
  constructor() {
    super();
    /*this.shadowRoot = */this.attachShadow({ mode: 'open' });
    this.shadowRoot.appendChild(template.content.cloneNode(true));
    this.$input = this.shadowRoot.querySelector('input');
  }

  static get observedAttributes() {
    return ['name', 'value', 'placeholder', 'disabled'];
  }

  attributeChangedCallback(name, oldValue, newValue) {
    this.$input.setAttribute(name, newValue);
  }

  get value() {
    return this.$input.value;
  }

  reset() {
    this.$input.value = ''; // resets the form without reloading the page
  }
}

customElements.define('chat-creation-input', ChatCreationInput);
