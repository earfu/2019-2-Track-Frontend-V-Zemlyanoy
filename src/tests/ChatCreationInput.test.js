import React from 'react';
import renderer from 'react-test-renderer';
import ChatCreationInput from '../components/ChatCreationInput';

it('renders correctly', () => {
  const component = renderer
    .create(
      <ChatCreationInput
        value="Test chat creation input text"
        onKeyPress={null}
        onSubmit={null}
        onChange={null}
      />,
    )
    .toJSON();
  // empty function used; a warning here is fine, function calls are tested in Storybook
  expect(component).toMatchSnapshot();
});
