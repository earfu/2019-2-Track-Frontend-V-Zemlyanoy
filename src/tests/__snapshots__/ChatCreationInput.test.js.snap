// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  className="chat-name-input"
  onChange={[Function]}
  onKeyPress={[Function]}
  onSubmit={[Function]}
  placeholder="Введите название"
  type="text"
  value="Test chat creation input text"
/>
`;
